.github/workflows/perf.yml

```diff
@@ -0,0 +1,69 @@
+name: perf
+
+# build on c/cpp changes or workflow changes
+on:
+  push:
+    paths:
+      - 'lib/**.[ch]'
+      - 'lib/**.cpp'
+      - 'src/**.[ch]'
+      - 'src/**.cpp'
+      - '**/CMakeLists.txt'
+      - 'cmake/Modules/**'
+      - 'util/buildbot/**'
+      - 'util/ci/**'
+      - '.github/workflows/**.yml'
+      - 'Dockerfile'
+      - '.dockerignore'
+# IMPORTANT: Do not run this workflow on pull request since this workflow
+# has permission to modify contents.
+
+permissions:
+  # deployments permission to deploy GitHub pages website
+  deployments: write
+  # contents permission to update benchmark contents in gh-pages branch
+  contents: write
+
+jobs:
+  # Current gcc version
+  perf_gcc_10:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install deps
+        run: |
+          source ./util/ci/common.sh
+          install_linux_deps g++-10
+
+      - name: Build
+        run: |
+          ./util/ci/build.sh
+        env:
+          CC: gcc-10
+          CXX: g++-10
+          CMAKE_FLAGS: "-DBUILD_BENCHMARKS=TRUE"
```

Comment:
also you can add `-DENABLE_GETTEXT=0 -DBUILD_SERVER=0` for free buildtime improvements

Reply:
done